src/sql-parser/src/parser.rs

```diff
@@ -3605,20 +3605,37 @@ impl<'a> Parser<'a> {
         };
         let in_cluster = self.parse_optional_in_cluster()?;
```

Comment:
Also, there's some opportunity for dedup in the parser, too https://github.com/MaterializeInc/materialize/pull/30260/files/93cb32bdaf21890e65f822711d62f0aec1556af4#diff-ff4ad701e3b476ee7c4794f262a310147cb7448f5da48c590ea9b5648aa89ff2

Reply:
I tried a more aggressive version of this where I pulled out a full prelude/postlude that was shared between all three forms, but IMO it felt a little clunky and didn't seem to get us a ton. Happy to defer to adapter folks though